--- linux-live-6.3.0-orig/initrd/cleanup	2007-11-13 16:11:40.000000000 +0100
+++ linux-live-6.3.0/initrd/cleanup	2010-07-31 17:08:58.010022000 +0200
@@ -48,17 +48,21 @@
 
 fumount $MOUNTDIR/*
 
+umount -a -r -d 2> /dev/null
+lo=$(losetup|cut -d: -f1); for d in $lo; do losetup -d $d; done
+
 # eject cdrom devices
-for MNT in $(ls -1 $MOUNTDIR 2>/dev/null); do
-   CD=$(cat /proc/sys/dev/cdrom/info 2>/dev/null | grep name | grep "$MNT")
-   if [ "$CD" -a "$MNT" ]; then
-      echo "Ejecting $MNT..."
-      /bin/eject -m /dev/$MNT >/dev/null 2>&1
-      echo "CD tray will be closed in 6 seconds..."
-      sleep 6
-      /bin/eject -t /dev/$MNT >/dev/null 2>&1
-   fi
-done
+if [ -e /bootdev ]; then
+  if [ "$(grep iso9660 /bootdev)" ]; then
+    bootcd=$(cut -d' ' -f1 /bootdev)
+    echo "Ejecting $bootcd..."
+    /bin/eject $bootcd 2>/dev/null
+    echo "CD tray will be closed in 10 seconds or hit enter..."
+    read -t 10
+    /bin/eject -t $bootcd 2>/dev/null
+  fi
+  mount -t proc none /proc && mount -o remount,rw / && rm -f /bootdev && mount -o remount,ro / && umount /proc
+fi
 
 if [ "$DEBUG_IS_ENABLED" ]; then
    echo "Everything should be correctly unmounted now so we can $1"
