--- linux-live-6.3.0-orig/tools/liblinuxlive	2009-02-10 12:42:05.000000000 +0100
+++ linux-live-6.3.0/tools/liblinuxlive	2010-07-31 17:06:15.463023398 +0200
@@ -255,7 +255,7 @@
    fi
 
    if [ "$LOCALE" != "" ]; then
-      cat /usr/share/locale/locale.alias | sed -r "s/#.*//" | egrep "$LOCALE|$LOCALE""_" | tail -n 1 | tr -s "[[:space:]]" " " | cut -d " " -f 2- | tr -d " "
+      echo "utf8"
    fi
 }
 
@@ -301,12 +301,12 @@
    if [ "$NOAUTO" = "" ]; then NOAUTO="auto"; fi
    if [ "$2" = "fstab" ]; then echo -n "$NOAUTO," ; fi
    if [ "$1" = "swap" ]; then echo "defaults,pri=1"; return 0; fi
-   echo -n "noatime,users,suid,dev,exec"
+   echo -n "noatime,suid,dev,exec"
 
    IOCHARSET=$(iocharset)
 
    if [ "$1" = "vfat" ]; then
-      echo -n ",quiet,umask=0,check=s,shortname=mixed"
+      echo -n ",quiet,umask=0,check=s,shortname=winnt,utf8"
       if [ "$IOCHARSET" ]; then
          echo ",iocharset=$IOCHARSET"
       fi
@@ -462,7 +462,7 @@
 {
    debug_log "create_module" "$*"
    rm -f "$2" # overwrite, never append to existing file
-   mksquashfs "$1" "$2" -b 256K -lzmadic 256K $3 $4 $5 $6 $7 $8 $9>/dev/null
+   mksquashfs "$1" "$2" -b 1M -comp lzma $3 $4 $5 $6 $7 $8 $9>/dev/null
    if [ $? -ne 0 ]; then return 1; fi
    chmod a-wx "$2" # remove execute and write attrib
    chmod a+r "$2" # add read for everyone
@@ -622,8 +622,11 @@
    debug_log "list_cdrom_devices" "$*"
    local CDDEVICE
 
-   for CDDEVICE in $(cat /proc/sys/dev/cdrom/info 2>/dev/null | head -n 3 | tail -n 1 | cut -d ":" -f 2); do
-      echo "/dev/$CDDEVICE"
+   for CDDEVICE in $(cat /proc/sys/dev/cdrom/info 2>/dev/null | grep 'drive name:' | cut -d ":" -f 2); do
+      dd if=/dev/$CDDEVICE of=/dev/null count=1 2>/dev/null
+      if [ $? -eq 0 ]; then
+         echo "/dev/$CDDEVICE"
+      fi
    done
 }
 
@@ -861,6 +864,7 @@
    echolog "starting linux filesystem support"
    modprobe_module ext2
    modprobe_module ext3
+   modprobe_module ext4
    modprobe_module reiserfs
    modprobe_module xfs
    modprobe_module vfat
