--- /etc/rc.d/rc.keymap	2009-09-16 02:54:02.481590671 +0200
+++ ../../../etc/rc.d/rc.keymap	2009-10-08 00:12:12.779135275 +0200
@@ -1,5 +1,8 @@
 #!/bin/sh
 # Load the keyboard map.  More maps are in /usr/share/kbd/keymaps.
 if [ -x /usr/bin/loadkeys ]; then
- /usr/bin/loadkeys -u fr-latin9.map
+  . /usr/lib/liblinuxlive
+  keyb="`cmdline_value keyb`"
+  [ -z "$keyb" ] && keyb=us
+  /usr/bin/loadkeys $keyb.map
 fi
