--- /etc/rc.d/rc.M	2010-07-23 13:35:53.000000000 +0200
+++ root/etc/rc.d/rc.M	2010-07-31 17:44:20.052022772 +0200
@@ -15,6 +15,11 @@
 # modified for colorized screen messages and local needs
 . /etc/shell-colors
 
+### modified for Live ###
+## LiveCD library
+. /usr/lib/liblinuxlive
+### /modified for Live ###
+
 # Tell the viewers what's going to happen.
 echo
 echo -e "${BOLDYELLOW}Going to multi user mode.${COLOR_RESET}"
@@ -124,15 +129,19 @@
   /bin/sh /etc/rc.d/rc.messagebus start
 fi
 
+### modified for Live ###
 # Start HAL:
-if [ -x /etc/rc.d/rc.hald ]; then
-  /bin/sh /etc/rc.d/rc.hald start
-fi
+#if [ -x /etc/rc.d/rc.hald ]; then
+#  /bin/sh /etc/rc.d/rc.hald start
+#fi
+### /modified for Live ###
 
+### modified for Live ###
 # Start Bluetooth:
-if [ -x /etc/rc.d/rc.bluetooth ]; then
-  /bin/sh /etc/rc.d/rc.bluetooth start
-fi
+#if [ -x /etc/rc.d/rc.bluetooth ]; then
+#  /bin/sh /etc/rc.d/rc.bluetooth start
+#fi
+### /modified for Live ###
 
 # Start wicd:
 if [ -x /etc/rc.d/rc.wicd ]; then
@@ -269,10 +278,15 @@
   . /etc/rc.d/rc.alsa
 fi
 
+#### modified for Live ###
 # Load a custom keymap if the user has an rc.keymap script.
-if [ -x /etc/rc.d/rc.keymap ]; then
-  . /etc/rc.d/rc.keymap
+#if [ -x /etc/rc.d/rc.keymap ]; then
+#  . /etc/rc.d/rc.keymap
+# Load custom preferences for Live session including keymap.
+if [ -x /etc/rc.d/rc.live ]; then
+  . /etc/rc.d/rc.live
 fi
+#### /modified for Live ###
 
 # Start the MySQL database:
 if [ -x /etc/rc.d/rc.mysqld ]; then
