--- /etc/rc.d/rc.M	2009-09-14 09:57:14.000000000 +0200
+++ ../../../etc/rc.d/rc.M	2009-10-05 23:42:15.000000000 +0200
@@ -10,9 +10,17 @@
 # Author:	Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
 #		Heavily modified by Patrick Volkerding <volkerdi@slackware.com>
 #
+### Modified for Live ###
+#
+
 
 [ ! -z `pidof -o %PPID /usr/sbin/splashy` ] &&	/usr/sbin/splashy_update "progress 60"
 
+### modified for Live ###
+## LiveCD library
+. /usr/lib/liblinuxlive
+### /modified for Live ###
+
 # Tell the viewers what's going to happen.
 echo "Going multiuser..."
 
@@ -319,6 +327,13 @@
   . /etc/rc.d/rc.keymap
 fi
 
+#### modified for Live ###
+# Start ZenLive extensions
+if [ -x /etc/rc.d/rc.live ]; then
+  . /etc/rc.d/rc.live
+fi
+#### /modified for Live ###
+
 # Start the MySQL database:
 if [ -x /etc/rc.d/rc.mysqld ]; then
   . /etc/rc.d/rc.mysqld start
