--- /etc/rc.d/rc.services	2009-10-24 10:40:42.000000000 +0200
+++ root/etc/rc.d/rc.services	2010-01-16 15:04:58.111694511 +0100
@@ -8,7 +8,7 @@
 	rc.messagebus rc.licences rc.scanluns rc.ntpd rc.bluetooth \
 	rc.dnsmasq rc.snmpd rc.lprng rc.atalk rc.saslauthd rc.sendmail \
 	rc.mysqld rc.httpd rc.openldap rc.samba rc.gpm rc.sysvinit \
-	rc.pcmcia rc.fuse rc.modules.local rc.modules-$(uname -r) ; do
+	rc.pcmcia rc.fuse rc.modules.local rc.modules-$(uname -r) rc.live ; do
 	echo $1 | grep $mask
 done
 }
