--- /etc/rc.d/rc.S	2010-07-23 13:35:53.000000000 +0200
+++ root/etc/rc.d/rc.S	2010-10-13 16:07:02.092621250 +0200
@@ -245,16 +245,20 @@
   read junk;
 fi # Done checking root filesystem
 
+### modified for Live ###
 # Any /etc/mtab that exists here is old, so we delete it to start over:
-/bin/rm -f /etc/mtab*
+#/bin/rm -f /etc/mtab*
 # Remounting the / partition will initialize the new /etc/mtab:
-echo -e "${BOLDCYAN}Creating /etc/mtab.${COLOR_RESET}"
-/sbin/mount -w -o remount /
+#echo -e "${BOLDCYAN}Creating /etc/mtab.${COLOR_RESET}"
+#/sbin/mount -w -o remount /
+### /modified for Live ###
 
+### modified for Live ###
 # Read in the correct / filesystem complete with arguments so mount will
 # show them correctly. This does not stop those arguments from functioning
 # but does prevent a small bug with /etc/mtab.
-/bin/grep ' / ' /proc/mounts | grep -v "^rootfs" > /etc/mtab
+#/bin/grep ' / ' /proc/mounts | grep -v "^rootfs" > /etc/mtab
+### /modified for Live ###
 
 # Fix /etc/mtab to list sys and proc if they were not yet entered in
 # /etc/mtab because / was still mounted read-only:
